lib/graphql/input_object_type.rb

```diff
@@ -27,6 +27,32 @@ def valid_non_null_input?(input)
       input_fields.all? { |name, field| field.type.valid_input?(input[name]) }
   end
 
+  def validate_non_null_input(input)
+    # Inputs are only required to provide an `all?` method and a `[]` method. That's not enough
```

Comment:
Ok, no prob! Do you think that it'd be OK to require that it be `Enumerable` and support `[]`?

My line of thinking is, the minimum requirement would be an `each` method. But at that point, it's a small step to also ask that it be `Enumerable`, since that's all you need for that anyway. So may as well just ask for it to be `Enumerable`.